### PR TITLE
Clean up unused dependencies and packaging issues

### LINKED ISSUE #9813: [Feature][all] Remove unused dependencies
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

now, the modules of 3.0-alpha have many unused dependencies, and the total install package size has up to 800M,  this operation will reduce the total size, I think this is very useful 

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by kezhenxu94**
Closes #9813 

distribution bin tar is ~476M, comparing to the previous ~679M

### CODE REVIEW

File: dolphinscheduler-data-quality/pom.xml [hunk 1]
```diff
@@ -171,34 +171,4 @@
...
-                            <mainClass>org.apache.dolphinscheduler.data.quality.DataQualityApplication</mainClass>
-                        </manifest>
-                    </archive>
-                </configuration>
-                <executions>
-                    <execution>
-                        <id>make-assembly</id>
-                        <phase>package</phase>
-                        <goals>
-                            <goal>single</goal>
```
Comment (caishunfeng) [inline dolphinscheduler-data-quality/pom.xml:197]: I remember that the developer added this to improve the packaging speed. Will it have any effect after removing it?
cc @zixi0825

File: dolphinscheduler-master/pom.xml [hunk 2]
```diff
@@ -67,11 +71,24 @@
...
         <dependency>
             <groupId>org.codehaus.janino</groupId>
             <artifactId>janino</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-common</artifactId>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
```
Comment (caishunfeng) [inline dolphinscheduler-master/pom.xml]: Is it still need to import this jar package with exclusion *?

Comment (kezhenxu94) [reply dolphinscheduler-master/pom.xml]: > Is it still need to import this jar package with exclusion *?

Master component only uses the class `ThreadUtil` in `Hadoop-common` itself, nothing in its transitive dependencies is used. note exclusion * only excludes everything `Hadoop-common` brings, not including `Hadoop-common` itself

File: dolphinscheduler-data-quality/pom.xml (hunk 1 above)
Comment (kezhenxu94) [reply dolphinscheduler-data-quality/pom.xml:197]: It doesn't make sense to me `jar-with-dependencies` packaging will improve the packaging speed, please provide more details @zixi0825

Comment (zixi0825) [reply dolphinscheduler-data-quality/pom.xml:197]: Jar-with-dependencies has nothing to do with improving packaging speed. I only use this to make the data quality task easier to use. When running the spark application, there is no need to add the paths of various drivers to reduce errors. I will mention PR to remove jar-with-dependencies.

Comment (kezhenxu94) [reply dolphinscheduler-data-quality/pom.xml:197]: > Jar-with-dependencies has nothing to do with improving packaging speed. I only use this to make the data quality task easier to use. When running the spark application, there is no need to add the paths of various drivers to reduce errors. I will mention PR to remove jar-with-dependencies.

@zixi0825 Thank you, that make sense to me. How do you plan to remove jar-with-dependencies, will that effect how users use this module?

Comment (zixi0825) [reply dolphinscheduler-data-quality/pom.xml:197]: > My idea is to add an attribute to the configuration file, fill in the list of jars that need to be used (provide default values), read the attribute and automatically add the configuration when running the Spark application, and it should have no effect on the user.

Comment (zixi0825) [reply dolphinscheduler-data-quality/pom.xml:197]: > > Jar-with-dependencies has nothing to do with improving packaging speed. I only use this to make the data quality task easier to use. When running the spark application, there is no need to add the paths of various drivers to reduce errors. I will mention PR to remove jar-with-dependencies.
> 
> @zixi0825 Thank you, that make sense to me. How do you plan to remove jar-with-dependencies, will that effect how users use this module?

My idea is to add an attribute to the configuration file, fill in the list of jars that need to be used (provide default values), read the attribute and automatically add the configuration when running the Spark application, and it should have no effect on the user.

**Review by caishunfeng — APPROVED** [review]
LGTM, it's a great job, thanks @kezhenxu94

**Comment by SbloodyS** [comment]
Wow. That's impressive.

**Comment by kezhenxu94** [comment]
WE can further reduce the package size if https://github.com/apache/dolphinscheduler/pull/9296 can get merged

**Pull request merged**